### PR TITLE
Update baklava

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
It's a merge from `develop` into `baklava-2.1.6`. I decided to do this in a branch and have a review because there are a lot of conflicts.

## Review hints

Files with conflicts:

- [x] Toggl.Core.Tests.Sync/Helpers/AppServices.cs
- [x] Toggl.Core.Tests/TestDependencyContainer.cs
- [x] Toggl.Core.Tests/UI/ViewModels/BaseViewModelTests.cs
- [x] Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs
- [x] Toggl.Core.Tests/UI/ViewModels/MainViewModelTests.cs
- [x] Toggl.Core.UI/Navigation/ViewModelLoader.cs
- [x] Toggl.Core.UI/ViewModels/EditTimeEntryViewModel.cs
- [x] Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs
- [x] Toggl.Core.UI/ViewModels/MainViewModel.cs
- [x] Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs
- [x] Toggl.Core/Analytics/BaseAnalyticsService.cs
- [x] Toggl.Core/Analytics/IAnalyticsService.cs
- [x] Toggl.Core/DependencyContainer.cs
- [x] Toggl.Core/Diagnostics/MeasuredOperation.cs
- [x] Toggl.Core/Interactors/IInteractorFactory.cs
- [x] Toggl.Core/Services/BackgroundService.cs
- [x] Toggl.Core/TogglSyncManagerFactory.cs
- [x] Toggl.Droid/Extensions/OnboardingExtensions.cs
- [x] Toggl.Droid/Fragments/MainFragment.Onboardings.cs
- [ ] Toggl.Droid/Fragments/MainFragment.cs
- [x] Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml
- [x] Toggl.Droid/Startup/AndroidDependencyContainer.cs
- [x] Toggl.Droid/Startup/TogglApplication.cs
- [ ] Toggl.Droid/Toggl.Droid.csproj
- [x] Toggl.Droid/packages.config
- [x] Toggl.iOS/Extensions/OnboardingExtensions.cs
- [x] Toggl.iOS/Startup/AppDelegate.Analytics.cs
- [x] Toggl.iOS/Startup/AppDelegate.cs
- [x] Toggl.iOS/Startup/IosDependencyContainer.cs
- [x] Toggl.iOS/Toggl.iOS.csproj
- [x] Toggl.iOS/packages.config

## :squid: Permissions
🙅‍♂ NO

### CODE REVIEW

File: Toggl.Droid/Fragments/MainFragment.cs [hunk 1]
```diff
@@ -182,12 +170,10 @@ public override void OnViewCreated(View view, Bundle savedInstanceState)
                  .DisposedBy(DisposeBag);
 
             ViewModel.TimeEntries
-                .ObserveOn(AndroidDependencyContainer.Instance.SchedulerProvider.MainScheduler)
```
Comment (daividssilverio) [inline Toggl.Droid/Fragments/MainFragment.cs]: I can't find the reason or specific commit where this was removed. Should this have been removed?

File: Toggl.Droid/Fragments/MainFragment.cs [hunk 2]
```diff
@@ -182,12 +170,10 @@ public override void OnViewCreated(View view, Bundle savedInstanceState)
...
                 .Subscribe(mainRecyclerAdapter.UpdateCollection)
                 .DisposedBy(DisposeBag);
 
             ViewModel.IsTimeEntryRunning
-                .ObserveOn(AndroidDependencyContainer.Instance.SchedulerProvider.MainScheduler)
```
Comment (daividssilverio) [inline Toggl.Droid/Fragments/MainFragment.cs]: Same for this one.

File: Toggl.Droid/Toggl.Droid.csproj [hunk 3]
```diff
@@ -56,7 +55,6 @@
     <WarningLevel>4</WarningLevel>
     <AndroidLinkSkip>
     </AndroidLinkSkip>
-    <AndroidLinkMode>SdkOnly</AndroidLinkMode>
```
Comment (daividssilverio) [inline Toggl.Droid/Toggl.Droid.csproj]: On #5685, Line 49 from this file has also been removed. That fixed our broken builds, didn't it? It's not necessary anymore?

Comment (daividssilverio) [reply Toggl.Droid/Toggl.Droid.csproj]: This doesn't seem to matter. Not to build at least.

Comment (juanlaube) [reply Toggl.Droid/Toggl.Droid.csproj]: This was removed in develop.

File: Toggl.Droid/Fragments/MainFragment.cs (hunk 1 above)
Comment (juanlaube) [reply Toggl.Droid/Fragments/MainFragment.cs]: In #5642, in the baklava branch, we replaced `SynchronizationContext.Current` for the `SchedulerProvider`.
In #5522 , in develop, the `ObserveOn` with `SynchronizationContext.Current` was removed.

I think this is correct 🤔

File: Toggl.Droid/Fragments/MainFragment.cs (hunk 2 above)
Comment (juanlaube) [reply Toggl.Droid/Fragments/MainFragment.cs]: Same thing here.

File: Toggl.Droid/Fragments/MainFragment.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Droid/Fragments/MainFragment.cs]: Oh! There you go! Thanks o/

**Pull request merged**